### ISSUE TITLE
Enable using object members in the custom hints

### ISSUE THREAD

**Issue by Krzysztof-Cieslak** [fsprojects/FSharpLint#235]
TBF, not sure if it's just not documented and I'm doing something wrong, or it's just not supported yet.

I'd like to define hints based on members usage, not only module functions. For example warning for `x.Value` (where x is an option)

**Comment by duckmatt** [comment]
Yep, currently not supported unfortunately. The hint matching is doing very little type checking at the moment, would be very nice to have type constraints like that in the hints for instance members
